### PR TITLE
Scopes

### PR DESCRIPTION
**Pull request by mrate**
Implementation of enhancement #875 Add scope context to search results

### CODE REVIEW

File: test/org/opensolaris/opengrok/analysis/java/JavaAnalyzerFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+import static org.opensolaris.opengrok.analysis.AnalyzerGuru.string_ft_nstored_nanalyzed_norms;
+import org.opensolaris.opengrok.analysis.Ctags;
+import org.opensolaris.opengrok.analysis.FileAnalyzer;
+import org.opensolaris.opengrok.analysis.Scopes;
+import org.opensolaris.opengrok.analysis.Scopes.Scope;
+import org.opensolaris.opengrok.analysis.StreamSource;
+import org.opensolaris.opengrok.analysis.c.CxxAnalyzerFactoryTest;
```
Comment (tarzanek) [inline test/org/opensolaris/opengrok/analysis/java/JavaAnalyzerFactoryTest.java]: This IS SPARTA!
I mean JavaAnalyzer test - remove this !

File: test/org/opensolaris/opengrok/analysis/java/JavaAnalyzerFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        };
+    }
+    
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        ctags = new Ctags();
+        ctags.setBinary(RuntimeEnvironment.getInstance().getCtags());
+
+        repository = new TestRepository();
+        repository.create(CxxAnalyzerFactoryTest.class.getResourceAsStream(
```
Comment (tarzanek) [inline test/org/opensolaris/opengrok/analysis/java/JavaAnalyzerFactoryTest.java]: same as above ... copy paste EVIL :)

File: testdata/sources/java/Sample.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        
+        public String InnerMethod() {
+            // somthing } */
+            /* }}} 
+                multi-line comment }{}
+            */
+            
+            System.out.print("I'm so useless");
+            
+            return "Why do robots need to drink?";
```
Comment (tarzanek) [inline testdata/sources/java/Sample.java]: was this class supposed to be called Marvin? :-D

File: web/default/style.css [hunk 4]
```diff
@@ -533,6 +533,7 @@ a.xmb	{ /* member */			color: #090; font-weight: bold; }
 a.xf	{ /* function */		color: #00f; font-weight: bold; }
 a.xmt	{ /* method */			color: #00f; font-weight: bold; }
 a.xsr	{ /* subroutine */		color: #00f; font-weight: bold; }
+a.scope { /* scope */                   color: dodgerblue; font-weight: bold; padding-left: 1ex; }
```
Comment (tarzanek) [inline web/default/style.css]: 2 more styles to be fixed !!!

File: testdata/sources/java/Sample.java (hunk 3 above)
Comment (mrate) [inline testdata/sources/java/Sample.java]: :-D Sort of, the return string is Bender's quote :-)

File: test/org/opensolaris/opengrok/analysis/java/JavaAnalyzerFactoryTest.java (hunk 2 above)
Comment (mrate) [inline test/org/opensolaris/opengrok/analysis/java/JavaAnalyzerFactoryTest.java]: Oh, my bad, repairing...

**Comment by tarzanek** [comment]
@dobrou let's merge this first for what @mrate  already has - then we will be glad to accept your patch! :)

@mrate - please fix my above comments and I will merge asap
Thank you guys! Very nice work.

**Comment by tarzanek** [comment]
thank you Tomas! 
merging

**Pull request merged**